chain/ethereum/src/chain.rs

```diff
@@ -291,92 +349,41 @@ impl Blockchain for Chain {
         self.adapter_selector
             .triggers_adapter(loc, capabilities, unified_api_version)
     }
-
-    async fn new_firehose_block_stream(
+    async fn new_block_stream(
         &self,
         deployment: DeploymentLocator,
-        block_cursor: FirehoseCursor,
+        store: Arc<dyn WritableStore>,
         start_blocks: Vec<BlockNumber>,
-        subgraph_current_block: Option<BlockPtr>,
         filter: Arc<Self::TriggerFilter>,
         unified_api_version: UnifiedMappingApiVersion,
     ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        let current_ptr = store.block_ptr();
+        if !self.chain_client().is_firehose() {
```

Comment:
`match self.chain_client()` is much better here since it protects against changes to that enum leading to the wrong behavior here